Source/DafnyCore/AST/TypeDeclarations/SubsetTypeDecl.cs

```diff
@@ -50,4 +50,48 @@ public enum WKind { CompiledZero, Compiled, Ghost, OptOut, Special }
   public virtual string GetDescription(DafnyOptions options) {
     return "subset type";
   }
+
+  public void AutoRevealDependencies(AutoRevealFunctionDependencies Rewriter, DafnyOptions Options, ErrorReporter Reporter) {
+    if (Witness is null) {
+      return;
+    }
+
+    var expressions = Constraint.SubExpressions.ToList().Concat(Witness.SubExpressions.ToList());
```

Comment:
Careful: Is SubExpressions taking ALL subexpressions or just the immediate sub-expressions?

Reply:
Ah good point. At some point I had implemented a method which iteratively collects subexpressions of a given expression (and its subexpressions' subexpressions and so on) and flattens them to a list. I should use that here.

Reply:
Done. Now it first flattens expressions to list of all subexpressions, then does the rest of the computation.